code/game/objects/structures/safe.dm

```diff
@@ -210,10 +284,31 @@ FLOOR SAFES
 		if(istype(I, /obj/item/clothing/accessory/stethoscope))
 			to_chat(user, "<span class='warning'>Hold [I] in one of your hands while you manipulate the dial!</span>")
 			return
+		else if(istype(I, /obj/item/thermal_drill))
+			if(drill)
+				to_chat(user, "There is already a drill attached!")
+			else if(do_after(user, 2 SECONDS, target = src))
+				if(!user.drop_item())
+					to_chat(user, "<span class='warning'>\The [I] is stuck to your hand, you cannot put it in the safe!</span>")
+					return
+				I.loc = src
+				drill = I
+				if(istype(drill, /obj/item/thermal_drill/diamond_drill))

```

Comment:
It would be more OOP to put this as a var on the drills themselves and then pull it from those.

Reply:
My first thought was to do that, but then I thought what if the drills are coded to be used on a second object.